indra/newview/llviewermessage.cpp

```diff
@@ -6667,7 +6671,21 @@ void process_script_teleport_request(LLMessageSystem* msg, void**)
             << LL_ENDL;
 
         instance->trackURL(sim_name, (S32)pos.mV[VX], (S32)pos.mV[VY], (S32)pos.mV[VZ]);
-        LLFloaterReg::showInstance("world_map", "center");
+        if (flags & BEACON_SHOW_MAP)
+        {
+            LLFloater* instance = LLFloaterReg::getInstance("world_map", "center");
+            if (instance)
+            {
+                bool old_auto_focus = instance->getAutoFocus();
```

Comment:
This probably won't be highlighting mLocationEditor as it should, I suggest:
               ```
 bool old_auto_focus = instance->getAutoFocus();
                instance->setAutoFocus(flags & BEACON_FOCUS_MAP);
                instance->openFloater("center");
                instance->setAutoFocus(old_auto_focus);
```

Reply:
That looks a lot cleaner.